### PR TITLE
Don't install plugins that aren't compatible with the installed jenkins

### LINKED ISSUE #82: Check core version needed for a plugin beforehand during update-status
During the plugin update, the jenkins_plugin_manager plugins.get_plugins method can check if the current jenkins core version is sufficient and raise an error if so.

This would allow a better user experience when trying to update with the latest version of the given plugin.
The update-status hook will then throw a jenkins_plugin_managerPluginMinCoreVersionError exception, which could be caught by the charm and stop there the update process (or trigger the update of the jenkins package if desired/available).

### PR DESCRIPTION
**Pull request by alejdg**
Closes #82.

### CODE REVIEW

File: unit_tests/test_plugins.py [hunk 1]
```diff
@@ -343,3 +346,57 @@ def test_broken_json_from_plugin_site(self, mock_restart_jenkins):
...
+        self.plugins._get_required_jenkins(plugin_name)
+        mock_get_plugin_info.assert_called_with(plugin_name)
+
+    @mock.patch("test_plugins.Plugins._get_plugins_to_install")
+    @mock.patch("charms.layer.jenkins.api.Api.get_plugin_version")
+    @mock.patch("test_plugins.Plugins._get_latest_version")
+    @mock.patch("test_plugins.Plugins._download_plugin")
+    def test_update_incompatible(self, mock_download_plugin, mock_get_latest_version, mock_get_plugin_version, mock_get_plugins_to_install, mock_restart_jenkins):
+        """
+        Make sure imcompatible plugins are not installed.
```
Comment (mthaddon) [inline unit_tests/test_plugins.py:362]: "incompatible"

File: unit_tests/test_plugins.py [hunk 2]
```diff
@@ -343,3 +346,57 @@ def test_broken_json_from_plugin_site(self, mock_restart_jenkins):
...
+        finally:
+            hookenv.config()["plugins-auto-update"] = orig_plugins_auto_update
+
+    @mock.patch("jenkins_plugin_manager.plugin.UpdateCenter.get_plugins")
+    @mock.patch("test_plugins.Plugins._get_required_jenkins")
+    @mock.patch("charms.layer.jenkins.api.Api.version")
+    def test__get_plugins_to_install(self, mock_jenkins_version, mock_get_required_jenkins, mock_get_plugins, mock_restart_jenkins):
+        """
+        When getting plugins to install all incompatible plugins will be
+        removed from the list an returned separatedly.
```
Comment (mthaddon) [inline unit_tests/test_plugins.py:383]: "and returned separately."

**Review by mthaddon — COMMENTED** [review]
Two very minor comments about comment typos, but otherwise looks good! Will approve once those are fixed.

File: unit_tests/test_plugins.py (hunk 1 above)
Comment (alejdg) [reply unit_tests/test_plugins.py]: dcdf96d

**Pull request merged**